io/io/test/TFileMergerTests.cxx

```diff
@@ -0,0 +1,48 @@
+#include "TFileMerger.h"
+
+#include "TMemFile.h"
+#include "TTree.h"
+
+#include "gtest/gtest.h"
+
+static void CreateATuple(TMemFile &file, const char *name, double value)
+{
+   auto mytree = new TTree(name, "A tree");
```

Comment:
to be (guaranteed to be) consistent, you may want to add
```
mytree->SetDirectory(&file);
````

Reply:
Hm... would that mean I will have to reset it in the end of that function?

Reply:
> Hm... would that mean I will have to reset it in the end of that function?

Why?  Because "file" in all your use case is already the current directory, adding SetDirectory would be a nop **however** if for somewhere 'file' is not the current directory at the time CreateATuple then the behavior of the function would *not* be what you expects. (it would attach the TTree to another file and thus the file.Write at the end would not store the TTree at alll).

And actually that reminds me we have a new constructor to use instead of (new + SetDirectory):
```
auto mytree = new TTree(name, "A tree", &file);
```

Reply:
Done.